pyro/contrib/gp/models/gpr.py

```diff
@@ -65,44 +65,38 @@ class GPRegression(GPModel):
         a covariance matrix to help stablize its Cholesky decomposition.
     :param str name: Name of this model.
     """
-    def __init__(self, X, y, kernel, noise=None, mean_function=None, jitter=1e-6,
-                 name="GPR"):
-        super(GPRegression, self).__init__(X, y, kernel, mean_function, jitter, name)
+    def __init__(self, X, y, kernel, noise=None, mean_function=None, jitter=1e-6):
+        super(GPRegression, self).__init__(X, y, kernel, mean_function, jitter)
 
-        noise = self.X.new_ones(()) if noise is None else noise
+        noise = self.X.new_tensor(1.) if noise is None else noise
         self.noise = Parameter(noise)
-        self.set_constraint("noise", torchdist.constraints.greater_than(self.jitter))
+        self.set_constraint("noise", torchdist.constraints.positive)
 
+    @autoname.scope(prefix="GPR")
     def model(self):
         self.set_mode("model")
 
-        noise = self.get_param("noise")
-
-        N = self.X.shape[0]
+        N = self.X.size(0)
         Kff = self.kernel(self.X)
-        Kff.view(-1)[::N + 1] += noise  # add noise to diagonal
+        Kff.view(-1)[::N + 1] += self.jitter + self.noise  # add noise to the diagonal
```

Comment:
because `noise` is positive with no lower bound, to avoid singularity, we have to add `jitter` here